packages/gatsby-plugin-sharp/src/__tests__/index.js

```diff
@@ -254,7 +254,7 @@ describe(`gatsby-plugin-sharp`, () => {
       })
 
       expect(result.presentationWidth).toEqual(20)
-      expect(result.presentationHeight).toEqual(10)
+      expect(result.presentationHeight).toEqual(20)
```

Comment:
This assertion change shows how this changes the behaviour - for example the input image here ( https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sharp/src/__tests__/images/144-density.png ) has aspectRatio of `1/2`. Previous code was preserving aspectRatio of input, current is not.

I don't know exactly what's correct way (preserving aspect ratio of image, or using aspectRatio calculated from `maxHeight/maxWidth`) TBH, but change like that seems dangerous to me. Especially when names of arguments are `maxWidth` and `maxHeight` so in my mind - this would limit size of image to be at most `maxWidth` / `maxHeight` which what I think currently is happening (?).

maybe we need additional switch for it if user wants to force aspect ratio calculated from `maxHeight/maxWidth`

Reply:
From only docs I found that say something about using both `maxWidth` and `maxHeight` ( https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-image/README.md#fluid-queries ):

> # Other options include maxHeight (set both maxWidth and maxHeight to crop),

So it seems like your change is actually what this document says. 

Something to check is to see if `fit` options (particularly `inside` variant) as any different than `cover`/`contain` - I think `inside` should return height 10, but `cover`/`contain` (one of those is default, don't remember which one) should actually crop and have 20?

So I think we should add another test case (or few) for `fit` that use different `fit` options